### PR TITLE
[MTSRE-415] Add optional `addons` property to mtbundles schema

### PR DESCRIPTION
**Pull request by MikeEdgar**
Proposal to add optional `addons` property to mtbundles schema

### CODE REVIEW

**Review by sugarraysam — CHANGES_REQUESTED** [review]

File: managedtenants/schemas/mtbundles.schema.yaml [hunk 1]
```diff
@@ -24,5 +24,24 @@ properties:
         $ref: "shared/sub_operators.json"
       subscriptionConfig:
         $ref: "shared/subscription_config.json"
+  addons:
```
[the remark this answers is not in the document]
Comment (MikeEdgar) [reply managedtenants/schemas/mtbundles.schema.yaml:19]: Done

**Comment by MikeEdgar** [comment]
This PR should be current now, but I can't see the build job to check on the failure.

**Comment by Ajpantuso** [comment]
> This PR should be current now, but I can't see the build job to check on the failure.

It was a [dependency issue](https://github.com/mt-sre/managed-tenants-cli/pull/92). This should be pass CI now.

File: managedtenants/schemas/mtbundles.schema.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ajpantuso) [reply managedtenants/schemas/mtbundles.schema.yaml:19]: @sugarraysam I don't think we need a full deprecation flow, but I'd prefer we order the changes like:
1. Only add the new `addons` property with this PR
2. Utilize just the refactors for `ci.py` from !245 so either `environments` or `addons` is accepted with `addons` preferred if available.
3. Migrate any necessary `config.yaml`s
4. Drop the `environment` property and branching code path in `ci.py`

Then feature is delivered at step 2 and `config.yaml` migrations are no longer a blocker, but just a cleanup task.

[the remark this answers is not in the document]
Comment (Ajpantuso) [reply managedtenants/schemas/mtbundles.schema.yaml:19]: Ah, then let's just merge this.

**Review by sugarraysam — APPROVED** [review]

**Pull request merged**